gym_ignition/utils/inverse_kinematics_nlp.py

```diff
@@ -0,0 +1,490 @@
+# Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+import os
+import numpy as np
+from enum import Enum, auto
+from typing import List, Dict, Optional, Union, NamedTuple
+
+
+class TargetType(Enum):
+    POSITION = auto()
+    ROTATION = auto()
+    TRANSFORM = auto()
```

Comment:
In my view, "POSE" would be more standard.

Reply:
It's robotics / iDynTree jargon :)

Reply:
Both are used in robotics in general.
Since iDynTree uses "pose" then of course just keep it for consistency.

Reply:
Actually, [iDynTree in this context is using Transform](https://robotology.github.io/idyntree/master/classiDynTree_1_1InverseKinematics.html#af97c2d5f44b64f7e422e70d20c0d49c9) and I'm using it for consistency reason.

Reply:
But I agree with you that `TargetType.POSE` is more intuitive for the user. I will change it.

Reply:
> Since iDynTree uses **"pose"** then of course just keep it for consistency.
 
Sorry, it was a typo, I meant that since iDynTree uses "transform" then complying with it is the best choice.

Reply:
To be honest, APIs exposed to the user should be as more clear as possible. Since we use `pose` also for the links, let's keep consistency with the rest of our APIs (you made me change my mind ^^).